SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/CodeGenerator/Unreal/InterfaceGenerator.cs

```diff
@@ -0,0 +1,197 @@
+using Improbable.Codegen.Base;
+using Improbable.CodeGen.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Improbable.CodeGen.Unreal
+{
+    public static class InterfaceGenerator
+    {
+        public static string ClassName = $"ExternalSchemaInterface";
+        public static string HeaderPath = $"ExternalSchemaInterface.h";
+        public static string SourceFile = $"ExternalSchemaInterface.cpp";
+
+        public static List<GeneratedFile> GenerateInterface(List<TypeDescription> componentTypes, Bundle bundle)
+        {
+            return new List<GeneratedFile>
+            {
+                new GeneratedFile(HeaderPath, GenerateInterfaceHeader(componentTypes, bundle)),
+                new GeneratedFile(SourceFile, GenerateInterfaceSource(componentTypes, bundle))
+            };
+        }
+
+        public static string GenerateInterfaceHeader(List<TypeDescription> componentTypes, Bundle bundle)
+        {
+            var builder = new StringBuilder();
+
+            builder.AppendLine($@"#pragma once
+
+#include ""CoreMinimal.h""
+#include ""Connection/SpatialWorkerConnection.h""
+#include ""Engine/NetDriver.h""
+#include ""SpatialConstants.h""
+#include ""SpatialDispatcher.h""
+#include ""SpatialNetDriver.h""
+#include ""HelperFunctions.h""
+
+{string.Join(Environment.NewLine, componentTypes.Select(component => $"#include \"{Types.TypeToHeaderFilename(component.QualifiedName)}\""))}
+
+#include <WorkerSDK/improbable/c_worker.h>
+
+DECLARE_LOG_CATEGORY_EXTERN(LogExternalSchemaInterface, All, All);
```

Comment:
```suggestion
DECLARE_LOG_CATEGORY_EXTERN(LogExternalSchemaInterface, Log, All);
```

It's very rare that we'd want `All` for Runtime default log verbosity. Was this intentional?

Reply:
Nope this was unintentional  👍